src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -2211,7 +2211,7 @@ static void add_retriable_send_initial_metadata_op(
                                    .grpc_previous_rpc_attempts);
   }
   if (GPR_UNLIKELY(calld->num_attempts_completed > 0)) {
-    grpc_mdelem retry_md = grpc_mdelem_from_slices(
+    grpc_mdelem retry_md = grpc_mdelem_from_slices_no_unref(
```

Comment:
I am not very familiar with the metadata code, but it looks like we didn't ref prior to this PR. Why should we change to `grpc_mdelem_from_slices_no_unref()` here?

Similar with most of the rest changes.

Reply:
That's a very good question. These are both static slices, and their ref/unref is a no-op.

Using no_unref() here, saves us 4 (vtable) function calls and two branches.